src/Components/Onboarding/Steps/Artists/ItemLink.tsx

```diff
@@ -49,26 +49,47 @@ interface Props extends React.HTMLProps<HTMLAnchorElement>, RelayProps {
   relay?: RelayProp
 }
 
+interface ArtistFollowProps {
+  id: string | null
+  __id: string
+}
+
 class ItemLink extends React.Component<Props, State> {
   state = {
     fadeIn: false,
     fadeOut: false,
     artist_id: this.props.artist.id,
   }
 
-  followArtist() {
-    const storeUpdater: SelectorStoreUpdater = (store, suggestedArtistData) => {
-      // Replace now followed artist in list with newly suggested artist
+  followArtist(artist: ArtistFollowProps) {
+    const storeUpdater = (store: RecordSourceSelectorProxy) => {
+      const mutationPayload = store.getRootField("followArtist")
+      const origArtist = mutationPayload.getLinkedRecord("artistEdge")
+
+      // Useful to import this directly for debugging
+      const localConnHandler = ConnectionHandler
+
+      // Somehow update the artist with the suggestion...
+      const selectedArtist = store.get(artist.__id)
+      const connection = localConnHandler.getConnection(origArtist, "ItemLink_suggested")
     }
 
     commitMutation(this.props.relay.environment, {
       mutation: graphql`
         mutation ItemLinkFollowArtistMutation($input: FollowArtistInput!) {
           followArtist(input: $input) {
-            artist {
-              # Get related artist (not this field that’s currently used!)
-              artists(size: 1) {
-                name
+            artistEdge {
+              node {
+                __id
+                related {
+                  suggested(first: 1) @connection(key: "ItemLink_suggested", filters: []) {
```

Comment:
We went thru a bunch of refactoring to make this a proper connection, that is actually accessible from within the mutation payload (nested under artist). It's disappointing that this refactor (while not being objectionable from a graph/shape standpoint), was pretty much for the express purpose of a Relay use....and we're still unable to use it in that fashion. What are we doing wrong?